### PR TITLE
polish and add more tests

### PR DESCRIPTION
**Pull request by rajadilipkolli**
## Summary by CodeRabbit

- **Refactor**
	- Simplified code structure in post management service.
	- Improved method naming conventions across services.
	- Enhanced error handling for post-related operations.

- **New Features**
	- Added method to update posts by username and title.
	- Introduced new test cases for post management scenarios.

- **Bug Fixes**
	- Improved error messages for post-related exceptions.
	- Added more robust handling of post retrieval and deletion.

- **Tests**
	- Expanded test coverage for post creation, retrieval, and deletion.
	- Added integration tests for edge cases in post management.
	- Introduced unit tests for deletion functionality in the post service.
	- Enhanced assertions to validate comments and tags in tests.
	- Added parameterized tests for various post operation configurations.
	- New tests for handling non-existent posts during retrieval and deletion.
	- Updated test methods to reflect changes in post attributes and behaviors.

### CODE REVIEW

**Comment by rajadilipkolli** [comment]
@coderabbitai, can you please help me to achieve below suggestion

```java
@ParameterizedTest
@MethodSource("postTestCases")
void shouldHandlePostOperationsWithDifferentConfigurations(
        String title, 
        List<PostCommentRequest> comments, 
        List<TagRequest> tags,
        boolean expectEmpty) {
    PostRequest postRequest = new PostRequest(
            title,
            "Test content",
            true,
            LocalDateTime.parse("2025-01-15T10:00:00"),
            comments,
            tags);
    
    // Test creation
    mockMvcTester.post()...
    
    // Test retrieval
    mockMvcTester.get()...
        .satisfies(postResponse -> {
            if (expectEmpty) {
                assertThat(postResponse.comments()).isEmpty();
                assertThat(postResponse.tags()).isEmpty();
            } else {
                // existing assertions
            }
        });
}

static Stream<Arguments> postTestCases() {
    return Stream.of(
        Arguments.of("emptyPost", List.of(), List.of(), true),
        Arguments.of("postWithComments", 
            List.of(new PostCommentRequest("title", "content", true, LocalDateTime.now())), 
            List.of(), 
            false),
        // more test cases
    );
}
```

**Comment by rajadilipkolli** [comment]
@coderabbitai, above suggested `shouldHandlePostOperationsWithDifferentConfigurations` works, which testcases should I remove?

File: jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/service/impl/JPAPostServiceImpl.java [hunk 1]
```diff
@@ -49,22 +49,24 @@ public PostResponse updatePostByUserNameAndTitle(PostRequest postRequest, String
                 .map(post -> {
                     log.debug("Updating post with title '{}' for user '{}'", title, userName);
                     this.postMapper.updateReferenceValues(postRequest, post);
-                    Post updatedPost = this.postRepository.save(post);
-                    return this.postMapper.postToPostResponse(updatedPost);
+                    return this.postRepository.save(post);
                 })
+                .map(this.postMapper::postToPostResponse)
```
Comment (rajadilipkolli) [inline jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/service/impl/JPAPostServiceImpl.java:54]: @coderabbitai, please help me to execute this line of code outside of transaction

Comment (rajadilipkolli) [reply jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/service/impl/JPAPostServiceImpl.java:54]: @coderabbitai, still `this.postMapper.postToPostResponse(updatedPost)` is under the method which is annotated with `@Transactional` so db connection is not released till this is executed

Comment (rajadilipkolli) [reply jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/service/impl/JPAPostServiceImpl.java:54]: @coderabbitai, annotating private method with @Tranasactional wont work.

Comment (rajadilipkolli) [reply jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/service/impl/JPAPostServiceImpl.java:54]: @coderabbitai, above code is same as creating a new transaction at method level, perform everything and then release the transaction when done. Instead of this, can we facade pattern so the fetch will be done in readonly Transaction, next two steps of updating and saving will be done under transaction and mapping will be out of transaction.

Comment (rajadilipkolli) [reply jpa/boot-jpa-jooq-sample/src/main/java/com/example/learning/service/impl/JPAPostServiceImpl.java:54]: @coderabbitai, please create an issue to implement the same
